src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/SupplementaryEvidenceCreatorTest.java

```diff
@@ -0,0 +1,131 @@
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.AUTH_DETAILS;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.SERVICE_TOKEN;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.USER_DETAILS;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.SampleData.USER_TOKEN;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SupplementaryEvidenceCreatorTest {
+
+    private static final String CASE_TYPE_ID = "Bulk_Scanned";
+    private static final String EVENT_TYPE_ID = "attachScannedDocs";
+
+    @Mock
+    private CcdAuthenticatorFactory authenticatorFactory;
+
+    @Mock
+    private CoreCaseDataApi coreCaseDataApi;
+
+    private SupplementaryEvidenceCreator creator;
+
+    @Before
+    public void setUp() {
+        given(authenticatorFactory.createForJurisdiction(any())).willReturn(AUTH_DETAILS);
+
+        creator = new SupplementaryEvidenceCreator(authenticatorFactory, coreCaseDataApi);
+    }
+
+    @Test
+    public void createSupplementaryEvidence_starts_and_submits_event() {
+        Envelope envelope = SampleData.envelope(2);
+        String eventToken = "token123";
+
+        StartEventResponse startEventResponse = mock(StartEventResponse.class);
+        given(startEventResponse.getToken()).willReturn(eventToken);
+
+        given(coreCaseDataApi.startEventForCaseWorker(any(), any(), any(), any(), any(), any(), any()))
+            .willReturn(startEventResponse);
+
+        creator.createSupplementaryEvidence(envelope);
+
+        verifyEventStarted(envelope);
+        verifyEventSubmitted(envelope, eventToken);
+    }
+
+    @Test
+    public void createSupplementaryEvidence_does_not_submit_event_when_starting_fails() {
+        Exception expectedException = new RuntimeException("test exception");
+
+        willThrow(expectedException)
+            .given(coreCaseDataApi)
+            .startEventForCaseWorker(any(), any(), any(), any(), any(), any(), any());
+
+        Throwable actualException = catchThrowable(
+            () -> creator.createSupplementaryEvidence(SampleData.envelope(1))
+        );
+
+        assertThat(actualException).isSameAs(expectedException);
+
+        verify(coreCaseDataApi, never())
```

Comment:
do we need this call?,
 do we need this test?

that call is very unlikely to throw a runtimeexception, 99.99999% of the time it would be a feign exception

Reply:
It might be too much. Will remove.